src/dev/so_migration/snapshot_plugin_types.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as cp from 'child_process';
+
+import {
+  extractMigrationInfo,
+  getMigrationHash,
+  SavedObjectTypeMigrationInfo,
+  // TODO: how to resolve this? Where to place this script?
+  // eslint-disable-next-line @kbn/imports/no_boundary_crossing
+} from '@kbn/core-test-helpers-so-type-serializer';
+import {
+  createTestServers,
+  createRootWithCorePlugins,
+  // TODO: how to resolve this? Where to place this script?
+  // eslint-disable-next-line @kbn/imports/no_boundary_crossing
+} from '@kbn/core-test-helpers-kbn-server';
+import { REPO_ROOT } from '@kbn/repo-info';
+import { ToolingLog } from '@kbn/tooling-log';
+
+import { mkdirp } from '../build/lib';
+
+type MigrationInfoRecord = Pick<
+  SavedObjectTypeMigrationInfo,
+  'name' | 'migrationVersions' | 'schemaVersions' | 'modelVersions'
+> & {
+  hash: string;
+};
```

Comment:
I'm assuming double the migration info's overall size isn't an issue? if so, we may want to also add the (already flattened in the summary) `mappings` to this structure, just in case we may want to compare them.

Reply:
yeah, I think the snapshot's scope is not an issue, we can definitely include more fields